src/grt/src/GlobalRouter.cpp

```diff
@@ -221,17 +221,27 @@ void GlobalRouter::saveCongestion()
   }
   std::ofstream out(congestion_file_name_);
 
-  std::vector<std::pair<GSegment, TileCongestion>> congestionGridsV,
-      congestionGridsH;
+  std::vector<std::tuple<GSegment, TileCongestion, std::vector<std::string>>>
+      congestionGridsV, congestionGridsH;
   fastroute_->getCongestionGrid(congestionGridsV, congestionGridsH);
   for (auto& it : congestionGridsH) {
+    GSegment seg;
+    TileCongestion tile;
+    std::vector<std::string> srcs;
+    std::tie(seg, tile, srcs) = it;
```

Comment:
This could be simplified with a structured binding

const auto& [seg, tile, srcs] = it;

same below